akka-javasdk/src/main/java/akka/javasdk/OriginAwareContext.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.javasdk;
+
+import akka.javasdk.annotations.Consume;
+import akka.javasdk.eventsourcedentity.EventSourcedEntity;
+import akka.javasdk.keyvalueentity.KeyValueEntity;
+
+import java.util.Optional;
+
+public interface OriginAwareContext extends Context {
```

Comment:
MessageContext (Consumer) and UpdateContext (Views) now extends `OriginAwareContext` which gives access to the `originRegion` and method `hasLocalOrigin`.